platform/src/ToolsManager.js

```diff
@@ -70,9 +132,25 @@ class ToolManager {
             }
 
             if (xhr.status === 200) {    
-                // Rewrite URLs in tool config
-                let baseURL = toolUrl.url.substring(0, toolUrl.url.lastIndexOf("/")); // remove the name of the json file (including the trailing slash)
-                let toolConfig = xhr.responseText.replaceAll("{{BASE-URL}}", baseURL);
+
+                var toolConfig;
+                let response_text =  xhr.responseText;
+                let response_text_regexp = response_text.match(new RegExp(/{{BASE-URL}}:[0-9]*/));
```

Comment:
```suggestion
                let response_text_regexp = response_text.match(new RegExp(/{{BASE-URL}}(:[0-9]+)?/));
```
Then loop over all matches and for each one decide individually what the correct replacement is.

Reply:
Regular expression was replaced as advised and now matches URL placeholders regardless of port presence.